kratos/utilities/mortar_utilities.h

```diff
@@ -375,10 +376,13 @@ class MortarUtilities
             normal += N[i_node] * Geom[i_node].GetValue(NORMAL); 
         }
         
-        if (norm_2(normal) > std::numeric_limits<double>::epsilon())
-        {
-            normal = normal/norm_2(normal); // It is suppossed to be already unitary (just in case)
-        }
+        const bool not_zero_vector = (norm_2(normal) > std::numeric_limits<double>::epsilon());
```

Comment:
1 - you are computing many time the norm_2

i would do

        const double mynorm = norm_2(normal)

     #ifdef KRATOS_DEBUG
     const bool not_zero_vector = (norm_2(normal) > std::numeric_limits<double>::epsilon());
     if (not_zero_vector == false) KRATOS_ERROR << "Zero norm normal vector. Norm:" << norm_2(normal) << std::endl;
     #endif
 
     normal/= mynorm;
     return normal;